src/libCrypto/EthCrypto.cpp

```diff
@@ -188,4 +210,121 @@ std::string ToUncompressedPubKey(std::string const& pubKey) {
   }
 
   return ret;
-}
\ No newline at end of file
+}
+
+secp256k1_context const* getCtx() {
+  static std::unique_ptr<secp256k1_context,
+                         decltype(&secp256k1_context_destroy)>
+      s_ctx{secp256k1_context_create(SECP256K1_CONTEXT_SIGN |
+                                     SECP256K1_CONTEXT_VERIFY),
+            &secp256k1_context_destroy};
+  return s_ctx.get();
+}
+
+// EIP-155 : assume the chain height is high enough that the signing scheme
+// is in line with EIP-155.
+// message shall not contain '0x'
+bytes RecoverECDSAPubSig(std::string const& message, int chain_id) {
```

Comment:
It'd be nice to have a unit test for some predefined input and known output (just to ensure it works in case someone needs to change it).

Reply:
sounds great - done